traffic_ops/traffic_ops_golang/dbhelpers/db_helpers.go

```diff
@@ -852,9 +852,33 @@ func TopologyExists(tx *sql.Tx, name string) (bool, error) {
 	return count > 0, err
 }
 
+// CheckTopologyV30 returns an error if the given Topology does not exist or if one of the Topology's Cache Groups is
+// empty with respect to the Delivery Service's CDN.
+func CheckTopologyV30(tx *sqlx.Tx, ds tc.DeliveryServiceNullableV30) (int, error, error) {
```

Comment:
This copy shouldn't be necessary -- all the "real work" of the DS API should be handled at the "v4 (latest) layer." I.e. the v1-3 handlers should all chain up to the v4 handler, where the request is actually handled, then converted back down to the handler at which the request came in.